src/pybind/mgr/nfs/export.py

```diff
@@ -659,7 +660,13 @@ def create_cephfs_export(self,
                              clients: list = [],
                              sectype: Optional[List[str]] = None) -> Dict[str, Any]:
 
-        check_cephfs_path(self.mgr, fs_name, path)
+        try:
+            check_cephfs_path(self.mgr, fs_name, path)
+        except cephfs.ObjectNotFound:
+            raise NFSObjectNotFound(f"either path {path} does not exist or"
```

Comment:
This does not make sense - If its an ObjectNotFound exception, then the exception should be thrown only when the directory is not found.

Reply:
yes the msg says path does not exist, also symlinks arent followed so it won't be able to look at it and return `cephfs.ObjectNotFound` in both the cases

Reply:
We aren't passing AT_SYMLINK_NOFOLLOW in flags, so how are symlinks not followed?

Reply:
It's not ideal to follow symlinks when it comes to NFS, I read it here https://docstore.mik.ua/orelly/networking_2ndEd/nfs/ch06_04.htm, (thanks to @mchangir for pointing it out).

So in helper we do 
`stx = fs_handle.statx(path.encode('utf-8'), cephfs.CEPH_STATX_MODE, 0)`

where the 3rd arg which is for `AT_SYMLINK_NOFOLLOW` is turned off

Reply:
> It's not ideal to follow symlinks when it comes to NFS, I read it here https://docstore.mik.ua/orelly/networking_2ndEd/nfs/ch06_04.htm, (thanks to @mchangir for pointing it out).

Sure - I know that.

> 
> So in helper we do `stx = fs_handle.statx(path.encode('utf-8'), cephfs.CEPH_STATX_MODE, 0)`
> 
> where the 3rd arg which is for `AT_SYMLINK_NOFOLLOW` is turned off

So, its `off`, which means symlinks are followed, isn't it?

Reply:
right

Reply:
In that case, you get a NotADirectoryError exception. The ObjectNotFound and the "might be a symlink" are confusing here.

Reply:
As following symlinks is turned off, the stat would fail on the provided symlink, thus the helper would raise cephfs.ObjectNotFound exception as it thinks the provided path is invalid, now here I can just put a stmt like "path does not exist" but that is a bit misleading. The path who's symlink does exist(created in the test case) and thus the symlink is valid too but just because the code is incapable of dealing with symlinks, it says symlink doesnt exist which is not true. that's why i thought of adding it in the stmt so that the user gets a hint that providing symlinks is discouraged

Reply:
> As following symlinks is turned off

Where is it turned off? Until AT_SYMLINK_NOFOLLOW is passed, symlinks are followed. So,

/path/to/symlink --> point to nothing, then stat would throw ENOENT. We should be showing "path is a symlink and is not allowed for export creation.

>the stat would fail on the provided symlink, thus the helper would raise cephfs.ObjectNotFound >exception as it thinks the provided path is invalid, now here I can just put a stmt like "path does not ?>exist" but that is a bit misleading.

This should be throwing "path is a symlink" irrespective of if its a broken symlink or not.

The path who's symlink is created does exist(created in the test case) and thus the symlink is valid too but just because the code is (intentionally made)incapable of dealing with symlinks, it says symlink doesnt exist which is not true.

We can infer its a symbolic link and throw the appropriate error message, isn't it?

Reply:
> > > As following symlinks is turned off
> > 
> > 
> > Where is it turned off?
> 
> [#51005 (comment)](https://github.com/ceph/ceph/pull/51005#discussion_r1163728017)
> 
> Until AT_SYMLINK_NOFOLLOW is passed, symlinks are followed. So,
> 
> > /path/to/symlink --> point to nothing, then stat would throw ENOENT. We should be showing "path is a symlink and is not allowed for export creation.
> > > the stat would fail on the provided symlink, thus the helper would raise cephfs.ObjectNotFound >exception as it thinks the provided path is invalid, now here I can just put a stmt like "path does not ?>exist" but that is a bit misleading.
> > 
> > 
> > This should be throwing "path is a symlink" irrespective of if its a broken symlink or not.
> > The path who's symlink is created does exist(created in the test case) and thus the symlink is valid too but just because the code is (intentionally made)incapable of dealing with symlinks, it says symlink doesnt exist which is not true.
> > We can infer its a symbolic link and throw the appropriate error message, isn't it?
> 
> it is raised here https://github.com/ceph/ceph/pull/51005/files#diff-66d1478f0663fee84e457bafb5b1fe683ad3701c4a30f92412e7df9a8acf5ccaR101
> 
> we cannot do something like stat.IS_SYMLINK, it would raise exception before it

We only need to be concerned about the path being a directory - anything else symlink, etc.. should error out saying path is not a directory. And if AT_SYMLINK_NOFOLLOW is used, there is enough information to deduce if the path is a directory or not (and obviously handle missing paths).

Reply:
alright, making the changes